ensemble_md/cli/run_EEXE.py

```diff
@@ -184,18 +211,22 @@ def main():
                 print('\nNote: No histogram correction will be performed.')
                 print('Note: No weight combination will be performed.')
 
-            # 3-5. Modify the MDP files and swap out the GRO files (if needed)
-            # Here we keep the lambda range set in mdp the same across different iterations in the same folder but swap out the gro file  # noqa: E501
-            # Note we use states (copy of states_) instead of states_ in update_MDP.
-            for j in list(range(EEXE.n_sim)):
-                os.mkdir(f'sim_{j}/iteration_{i}')
-                MDP = EEXE.update_MDP(f"sim_{j}/iteration_{i - 1}/{EEXE.mdp.split('/')[-1]}", j, i, states, wl_delta, weights, counts)   # modify with a new template  # noqa: E501
-                MDP.write(f"sim_{j}/iteration_{i}/{EEXE.mdp.split('/')[-1]}", skipempty=True)
-                # In run_EEXE(i, swap_pattern), where the tpr files will be generated, we use the top file at the
-                # level of the simulation (the file that will be shared by all simulations). For the gro file, we pass
-                # swap_pattern to the function to figure it out internally.
-        else:
-            swap_pattern = None
+        # Placeholder helper function.
+        # In the current code, there is one ensemble member per rank.
+        def each_ensemble_member():
```

Comment:
This is far from normative, but it would be very helpful to distinguish between
* things that need to be done once for each ensemble member (and which only affect dependent tasks for that member)
* things that have different resource requirements than the simulator (such as only a single process) but which have dependencies or dependents on multiple ensemble members.